src/backend/partaj/core/api/referral_attachment.py

```diff
@@ -0,0 +1,114 @@
+"""
+Referral attachment related API endpoints.
+"""
+from django.http import Http404
+
+from rest_framework import viewsets
+from rest_framework.permissions import BasePermission
+from rest_framework.response import Response
+
+from .. import models
+from ..serializers import ReferralAttachmentSerializer
+from .permissions import NotAllowed
+
+
+class UserIsReferralRequester(BasePermission):
+    """
+    Permission class to authorize a referral answer's author on API routes and/or actions for
+    objects with a relation to the referral answer they created.
+
+    NB: we're using `view.get_referralanswer()` instead of `view.get_object()` as we expect this to
+    be implemented by ViewSets using this permission for objects with a relation to a referral.
+    """
+
+    def has_permission(self, request, view):
+
+        referral = view.get_referral(request)
+        return (
+            request.user.is_authenticated
+            and referral.users.filter(users__id=request.user.id).exists()
```

Comment:
I found the issue in the referral attachment test: you're looking up a `users` field on user objects, which do not have it.
```suggestion
            and referral.users.filter(id=request.user.id).exists()
```